File: lib/layer2_core/dark/e_surface_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd3;
      case 'standard': return cstd3;
      case 'blue': return cbd3;
      case 'green': return cgd3;
      case 'red': return crd3;
      case 'nothing': return cnod3;
      default: return cstd3;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd3
            AppTheme.STANDARD -> R.color.cstd3
            AppTheme.BLUE -> R.color.cbd3
            AppTheme.GREEN -> R.color.cgd3
            AppTheme.RED -> R.color.crd3
            AppTheme.NOTHING -> R.color.cnod3
        }
    }
}
